src/python/pants/engine/fs.py

```diff
@@ -39,6 +39,18 @@ class Snapshot:
     dirs: Tuple[str, ...]
 
 
+@dataclass(frozen=True)
+class FileListing:
```

Comment:
Totally open to other names. Here are the options I thought of:

* FileListing
* SnapshotListing
* FilesAndDirs
* ResolvedGlobs
* PathStats

Reply:
`Paths` might be good. `PathGlobs` expand to `Paths`.